packages/provider/src/tasks/tasks.ts

```diff
@@ -164,6 +167,138 @@ export class Tasks {
         })
     }
 
+    /**
+     * @description Generates a PoW Captcha for a given user and dapp
+     *
+     * @param {string} userAccount - Dapp User address
+     * @param {string} dappAccount - Dapp User address
+     */
+    async getPowCaptchaChallenge(userAccount: string, dappAccount: string, origin: string): Promise<PoWCaptcha> {
```

Comment:
There should be some limitation on getting challenges. No challenge should be returned if the dapp does not have the POW product.

Reply:
Same goes for any of the other API calls relating to POW. Checks basically want to be

- does the app have POW?
- is the origin in the request the same as the registered dapp origin in the contract

Don't implement the origin check for now.

Reply:
To be resolved in 

https://github.com/orgs/prosopo/projects/4/views/1?pane=issue&itemId=54757049
https://github.com/orgs/prosopo/projects/4/views/1?pane=issue&itemId=54757136

That is unless you want the feature checking in this pr? Relates to https://github.com/orgs/prosopo/projects/4/views/1?pane=issue&itemId=54753652, which I think is a v high priority follow up pr imo